### PR TITLE
Fix leak in LandmarkCostFunctionTest.

### PR DESCRIPTION
**Pull request by gaschler**

### CODE REVIEW

File: cartographer/mapping_3d/pose_graph/landmark_cost_function_test.cc [hunk 1]
```diff
@@ -66,6 +66,7 @@ TEST(LandmarkCostFunctionTest, SmokeTest) {
                           jacobians_ptrs.data());
   EXPECT_THAT(residuals, ElementsAre(DoubleEq(1.), DoubleEq(0.), DoubleEq(0.),
                                      DoubleEq(0.), DoubleEq(0.), DoubleEq(0.)));
+  delete cost_function;
```
Comment (pifon2a) [inline cartographer/mapping_3d/pose_graph/landmark_cost_function_test.cc]: Let's just make it an std::unique_ptr above.

**Comment by ojura** [comment]
Why not use a unique pointer?

File: cartographer/mapping_3d/pose_graph/landmark_cost_function_test.cc (hunk 1 above)
Comment (gaschler) [reply cartographer/mapping_3d/pose_graph/landmark_cost_function_test.cc]: Done, PTAL

**Comment by gaschler** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user gaschler. Merge queue now has a length of 2.

**Pull request merged**